apps/api/src/common/error-exception.filter.ts

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { GenericException } from './../common/generic-exception';
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+  HttpStatus,
+  Logger,
+  Inject,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { FailedResponse } from '../common/ro/failed-response.ro';
+import { ClassValidationParser } from '../common/parsers/class-validation.parser';
+import { CommonError } from 'src/common/common.errors';
+
+@Catch(Error)
+export class ErrorExceptionFilter implements ExceptionFilter {
+  constructor(@Inject(Logger) private readonly logger: Logger) {}
+
+  /**
+   * Transform a generic thrown exception to a `FailedResponse`
+   *
+   * @param exception: An exception caught by the handler
+   * @returns A failed response object
+   */
+  transformHttpException(exception: Error): FailedResponse {
+    const exceptionMessage: any = exception.message;
+
+    return {
+      errorType:
+        exceptionMessage.error ||
+        (exception as any).response.error ||
```

Comment:
Nit, but is it possible to add a type here?

Reply:
I really want to get some type for an exception, but it is really hard to be specific as exceptions can be trigger from any part of the stack.

Something we can improve over time?

Reply:
Totally! 👍